### PR TITLE
[Enhancement] Uses the value in "feature_image" as the default OpenGraph image, if "image" was not stated

### PR DESCRIPTION
**Pull request by ManuelFte**
This update makes the theme take the value in `feature_image` as the default OpenGraph image, unless it's overridden in the front matter by the `image` variable, either locally or globally.

Currently, the user has to explicitly state an OpenGraph image with the `image` variable, and the theme will not show any if not, even if the page has a featured image.

### CODE REVIEW

**Comment by daviddarnes** [comment]
This is an interesting addition, however what if I don’t want to add an open graph image but I want a feature image? Having them split means I can change to a different Twitter card type without loosing a feature image on the page 🙂

**Comment by ManuelFte** [comment]
`image` overrides `feature_image` when both are present, so in order to have `feature_image` without an OpenGraph image, one could simply set the value of `image` to an empty string:

```
feature_image: /images/myimage.png
image: ""
```

**Comment by daviddarnes** [comment]
I get what the benefit is, but it’s not really expected behaviour. It causes a counter-intuitive action if I don’t want an open graph image, I have to set it to remove it.
It’ll also break existing sites as the theme will update and apply the feature image regardless if they wanted that or not.

**Comment by daviddarnes** [comment]
Also realised that the `image` frontmatter option comes directly from the SEO plugin. Passing in the feature image as a fallback will need more than a single meta tag, and adding those will overwrite a fair chunk of what the SEO plugin is already handling pretty well 🙂

**Comment by ManuelFte** [comment]
I understand that in the case of a user who wants a featured image but no OpenGraph image this update would require to turn the setting off, but in my opinion this sounds like a very edge case, and this update targets the most common use case, which is when the user does want the featured image to also be the OpenGraph image.

It's customary for blogs and websites in general to use the featured image as OpenGraph image, and most themes for Jekyll and other platforms such as WordPress do this automatically and don't even make a distinction between the two. Since in Alembic these features are split, and there's no default behavior for any, the user who does want to do this has to set the image twice, resulting in duplication in all posts. This update allows to set it only once, which is what the user most likely expects, rather than repeating it each time.

I also want to highlight that this is only the default behavior. The features are still split, and whoever wants the OpenGraph image to be different can still override it the same way it works currently. Whoever wants no OpenGraph image at all can still disable this behavior, either by post or globally by setting an empty `image` in the `_config.yml`.

It's true that this'd make an OpenGraph image appear for users who left it empty, but as I mentioned, this sounds like a very rare case, and even in that situation it'd hardly be an inconvenience. I don't see what harm can an OpenGraph image do, and I wonder how many users are leaving the variable empty voluntarily because for some reason they don't want any OpenGraph image, or simply because they don't know there are two different variables (like me, who was baffled that the theme was not using my featured image as OpenGraph image until I checked the code and saw the reason, prompting me to submit this PR).

There are then 3 possible cases:

- If someone wants to use the featured image as OpenGraph image (which is the most likely case), this update makes it easier for them. Currently the only way is to set it twice for each post.
- For those who want the images to be different, the behavior hasn't changed. They can still set different values as it works now.
- If someone wants featured image but no OpenGraph image, they can easily disable it by setting an empty `image` in `_config.yml`.

In other words, for the most common use case it makes easier something that currently is inconvenient, and for the least common one it's also easy to turn it off (and certainly it's much less work than what it takes to continue adding each image twice).

**Comment by ManuelFte** [comment]
As for Jekyll SEO, it doesn't really interfere because that plugin only sets the OpenGraph image when `image` has a value, while this feature only sets it when it doesn't have a value; therefore each feature only works when the other isn't doing rather than both at once.

PS: Sorry for the wall of text, didn't realize I had written so much, lol.

**Comment by daviddarnes** [comment]
I really appreciate the effort you've gone into for this suggestion @ManuelFte, but I don't really want to mix these two options together. You're right that there's often cases where the feature image matches the Twitter card image, however the SEO `image` and `feature_image` serve different purposes.

The `feature_image` is purely presentational, appearing at the top of the page to make the page more aesthetically pleasing and providing an interesting backdrop to the feature text (if applied).

The `image` is for SEO, and will be used as part of social media previews and in JSON-LD data. Both are images, and are for visual appeal, but are shown in different context. Once more they are cropped very differently, and mixing the two may cause people to accidentally serve a badly cropped image in another context.

Hope you understand. Note that you're free to do this in your own fork! The addition you made looks very small and I can see that being added to the `site-before-start.html` include without much trouble 👍
